0028-web-monetization/0028-web-monetization.md

```diff
@@ -41,33 +41,39 @@ Web Monetization makes use of [SPSP](../0009-simple-payment-setup-protocol/0009-
-  - Each time a packet with a nonzero amount is fulfilled, the provider notifies the browser, and the browser emits an event on `document.monetization`. The event's type is `monetizationprogress`. The event has a `detail` field containing the details of the packet ([specified below](#monetizationprogress)). If there are no listeners the event MAY NOT be emitted.
+- The user's agent navigates to a webpage.
+- The user's agent sets `document.monetization` to an Object which implements [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget).
+- The user's agent sets `document.monetization.state` to `pending`.
+- The user's agent observes the `<head>` and looks for a Web Monetization `<meta name="monetization">` tag ([specified below](#meta-tags-set)).
+  - There MUST be only one `<meta name="monetization">` tag at any given time
+  - The `<meta>` Tags Set MUST be in the `<head>` of the document.
+  - The `<meta>` Tags Set MUST be in the top level window (i.e. not inside an iframe)
+
+- Below is repeated for every semantically (consider `meta.content = meta.content`) new  `<meta name="monetization">` tag seen for the life of the page:
+  - If any of the Web Monetization `<meta>` Tags Set are malformed, the browser will stop here. The user's agent SHOULD report a warning via the console.
+  - If the Web Monetization `<meta name="monetization">` tag is well-formed, the browser should extract the Payment Setup Endpoint ([Payment Pointer](../0026-payment-pointers) or [SPSP](../0009-simple-payment-setup-protocol) Url).
+  - The user's agent generates a fresh **random** UUID (version 4) and uses this as a Monetization ID. This MUST be regenerated upon dynamic changes to the meta tags and MUST be unique per page load in order to avoid user tracking.
+  - The user's agent passes these payment details to the user's provider (see [Payment Handler Flow](#payment-handler-flow)).
+  - The provider resolves the Payment Setup Endpoint and begins to pay. The provider MAY be acting from a different machine from the user. Cookies and session information MUST NOT be carried with any requests made to resolve the Payment Setup Endpoint, with the exception of the Monetization ID.
+    - On the SPSP query to resolve the Payment Setup Endpoint, a `Web-Monetization-Id` header is sent, containing the Monetization ID. The server running the web-monetized site may use this to associate future requests by the user with their payments.
+    - With the `destination_account` and `shared_secret` fetched from the SPSP query, a STREAM connection is established. A single STREAM is opened on this connection, and a positive SendMax is set.
+    - The provider SHOULD set their SendMax high enough that it is never reached, and make payment adjustments by limiting throughput.
+  - Once the STREAM connection has fulfilled an ILP packet with a non-zero amount, the provider notifies the browser:
```

Comment:
@sharafian @traviscrist 

The phrase:  "the provider notified (grammar error?) the browser, and the browser dispatches an event on `document.monetization`."  kind of threw me for bit, as I misread it as "provider notifies browser VIA dispatching an event on document.monetization"

In terms of "the provider notified the browser":

The oauth polyfill just "does it" and the "provider" in the case of the extension sends a wext runtime 'monetizationProgress' message to the content script.

Reply:
This is the as-yet-undefined interface from provider to browser that we need to  design and implement more explicitly.

I don't think we need to do that in this PR but we should start thinking about soon and include in the W3C proposal.

Reply:
Take a crack :) 
I might have a go tomorrow or the weekend. 
Sleepy McJoe today :)